geomstats/geometry/spd_matrices.py

```diff
@@ -422,7 +399,7 @@ def inner_product(self, tangent_vec_a, tangent_vec_b, base_point):
                 (gs.maximum(n_tangent_vecs_a, n_tangent_vecs_b), 1, 1))
 
         if power_affine == 1:
-            inv_base_point = gs.linalg.inv(base_point)
+            inv_base_point = SymmetricMatrices.inv(base_point)
```

Comment:
Not so sure because this inv method is valid for symmetric matrices in GLn, which is a larger set than SPD (negative eigen values are allowed). GLn already also has an inv method. The question is whether this one is more efficient with the eigenvalue decomposition...

Reply:
> The question is whether this one is more efficient with the eigenvalue decomposition...
So it turns out using the SVD for exp and log is highly beneficial over the gs.linalg implementations, but it is not for the inverse, so I should remove this method maybe?

Reply:
Problem solved then, thanks for checking!